Create Player#sendActionBar with components as parameters also pass t…

JanTuck opened the pull request:
…hrough legacy sendActionBar(String) to fix some miscolouring


To see the issue try using Player#sendActionBar(ChatColor.of(Color.RED) + "Test")

This will horribly turn all black, this pr fixes that and also adds a method to use with components since the other one is deprecated for some reason.

JanTuck committed:
Create Player#sendActionBar with components as parameters also pass through legacy sendActionBar(String) to fix some miscolouring

JanTuck committed:
Fix call

Proximyst commented on Spigot-Server-Patches/0544-Create-Player-sendActionBar-with-component-support.patch:
```diff
@@ -0,0 +1,34 @@
++    @Override
++    public void sendActionBar(BaseComponent... components) {
++        if (getHandle().playerConnection == null || components == null || components.length == 0) return;
++        PacketPlayOutChat packetPlayOutChat = new PacketPlayOutChat(null, net.minecraft.server.ChatMessageType.GAME_INFO, SystemUtils.getNullUUID());
```
Why do you send `null` instead of the supplied components?

JanTuck replied:
Supplying null was correct, but I forgot to set the components in the pr I sent, thank you for noticing.

JanTuck committed:
Fix proper setting components

I tested it before, but reverted files cause my IJ really wants to reorder the imports.

aikar commented on Spigot-API-Patches/0218-Create-Player-sendActionBar-with-component-support.patch:
```diff
@@ -0,0 +1,28 @@
++     *
++     * @param components The message to send in components
++     */
++    public void sendActionBar(@NotNull net.md_5.bungee.api.chat.BaseComponent... components);
```
We don't need this, an API exist that works just fine for it.

Previously I marked deprecated on the component API because there was zero reason to use components there.

We can simply remove that deprecation in string based action bar api patch instead.

now, adding an API only patch for .sendActionBar like you have here that simply fowards to .sendMesage(ChatMessageType.ACTION_BAR, comps); would be ok as a utility method, but this should all just be merged into my original patch.
